src/Interface/HasAggregate.agda

```diff
@@ -0,0 +1,9 @@
+{-# OPTIONS --safe --cubical-compatible #-}
+
+module Interface.HasAggregate where
+
+record HasAggregate (A : Set){B : Set} : Set where
+  infixl 7 _+_
+  field _+_ : A → B → A
```

Comment:
This feels quite unintuitive to me and also could easily create ugly type-inference behaviour. 

What about making `Epoch` simply definitionally equal to `ℕ` and thus only requiring the more intuitive `HasAdd` typeclass, which btw I would model as follows so as to reuse Semigroup:
```agda
record HasAdd {a} (A : Set a) where
  field {{ sg }} : Semigroup A
  infixl 7
  _+_ = _<>_
open HasAdd ⦃...⦄ public
```

Reply:
One problem about this is that `Epoch` is actually a `Maybe Natural` (I think?) in the Haskell code, since the genesis epoch is special. I personally think that's a bit stupid, but apparently Haskellers like that sort of stuff.

So in that sense, `Epoch` is more of an `ℕ`-module, with this addition being the `ℕ`-action. Depending on your viewpoint, you can also see `Epoch` as being `ℕ ∪ {-1}`. But the standard symbol for an action is the multiplication symbol, which might be more confusing... 🤔